akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1067,17 +1061,6 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
       }
     }
 
-    // is called from the owning [[GraphStage]]
-    private[stage] def onStop(outstandingPromises: Set[Promise[Done]]): Unit = {
-      if (outstandingPromises.nonEmpty) {
-        val detachedException = streamDetatchedException
-        val iterator = outstandingPromises.iterator
-        while (iterator.hasNext) {
-          iterator.next().tryFailure(detachedException)
-        }
-      }
-    }
-
     private def onAsyncInput(event: T, promise: OptionVal[Promise[Done]]) = {
       interpreter.onAsyncInput(GraphStageLogic.this, event, promise, handler.asInstanceOf[Any ⇒ Unit])
```

Comment:
This is where the `OptionVal[Promise[Done]]` actually got an instance of `OptionVal` instance when I looked at the generated byte code, because of `onAsyncInput` being a `Function` value and not a method. Maybe good to fix that as well, even if it didn't show anything much for me?

Reply:
I'm not sure these things really matter right now. That async dispatch will create a message and send it to an actor which creates so much more garbage that an Option will probably not even be measurable.

Reply:
Fair point